packages/block-library/src/freeform/edit.js

```diff
@@ -43,6 +43,10 @@ export default function ClassicEdit( {
 	onReplace,
 } ) {
 	const { getMultiSelectedBlockClientIds } = useSelect( blockEditorStore );
+	const canRemove = useSelect(
+		( select ) => select( blockEditorStore ).canRemoveBlock( clientId ),
```

Comment:
The `canRemoveBlock` selector considers the template lock, so we can close #29743.